src/components/ingestion/__test__/HttpIngestionEngine.unit.test.js

```diff
@@ -0,0 +1,49 @@
+import nock from 'nock'
+import HttpIngestionEngine from '../HttpIngestionEngine'
+
+describe('HttpIngestionEngine', () => {
+	describe('constructor', () => {
+		it('should throw an error when called without a URL', () => {
+			expect(() => {
+				new HttpIngestionEngine() // eslint-disable-line no-new
+			}).toThrow(Error)
+		})
+	})
+
+	describe('getInputStream', () => {
+		it('it should stream the client request body', (done) => {
+			const body = 'it worked!'
+
+			nock('http://example.com').get('/test.ts').reply(200, body)
+
+			const fileEngine = new HttpIngestionEngine('http://example.com/test.ts')
+
+			const inputStream = fileEngine.getInputStream()
+
+			inputStream.on('data', (data) => {
+				expect(data.toString()).toEqual(body)
+				done()
+			})
+		})
+
+		it('it should pass on errors correctly', (done) => {
+			const message = 'oh noes!?'
+			const url = 'http://example.com/error-test.ts'
+
+			nock('http://example.com').get('/error-test.ts').replyWithError({
+				message,
+			})
+
+			const fileEngine = new HttpIngestionEngine(
+				url,
+			)
+
+			const inputStream = fileEngine.getInputStream()
+
+			inputStream.on('error', (err) => {
+				expect(err.message).toEqual(`request to ${url} failed, reason: ${message}`)
+				done()
```

Comment:
If no error is sent this test will fail but it will fail based on timeout settings, right?  I would prefer a more explicit test that will fail directly if no error has been thrown (who knows if we will have to play with test timeout settings in some environment some day).

Would it be reasonable to modify this test so the `error` event sets a value and the `finish` event is the place that actually runs the `expect` statements in terms of those values?  For instance, having `error` append the error to an `errorCache` array, and then in `finish` have something like:

`expect(errorCache.length).toBe(1)`

Reply:
I looked into this, but problem I believe I was having that it didn't emit finish. My research at the time indicated that the error event ended the stream, emitting nothing else, and this was expected and normal. But I'll check again.

Reply:
Okay, used Jest's promise resolution/rejection support for this. Seems better.